pkg/controllermanager/controller/seed/seed_garden_secret_mapper.go

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package seed
+
+import (
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	"github.com/gardener/gardener/pkg/logger"
+
+	corev1 "k8s.io/api/core/v1"
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func filterGardenSecret(obj interface{}) bool {
+	secret, ok := obj.(*corev1.Secret)
+	if !ok {
+		return false
+	}
+	if secret.Namespace != v1beta1constants.GardenNamespace {
+		return false
+	}
```

Comment:
Can't we already restrict the shared informer factory to only watch the `garden` namespace?

Reply:
The [shared informer](https://github.com/gardener/gardener/blob/d8eea89323d880440d9f1034e4b8f2403f72d947/pkg/controllermanager/controller/seed/seed.go#L109) also includes `gardener.cloud/role` secrets from other namespaces as we cache them for [update calls](https://github.com/gardener/gardener/blob/d8eea89323d880440d9f1034e4b8f2403f72d947/pkg/controllermanager/controller/seed/seed_reconcile.go#L169).

Reply:
I didn't get it 😢 Can you explain more?

Reply:
Sorry, my bad. I meant that we also need to list secrets from other namespace for cleaning them up, see [here](https://github.com/gardener/gardener/blob/d8eea89323d880440d9f1034e4b8f2403f72d947/pkg/controllermanager/controller/seed/seed_reconcile.go#L120). But you got me thinking: maybe we can use a [metadata-only watch](https://github.com/kubernetes-sigs/controller-runtime/pull/1174) for this so save memory.

Reply:
I tried out the metadata only watch in ce0203cdd49eb6f455084d87f487cf09cf526145 which worked quite well. Nevertheless, I'd like analyse the situation with the help of profiling and monitoring to see if it's really worth. I already pushed it so you can have a look and provide feedback if you want (tests haven't been adjusted yet).

Reply:
After some debugging I compiled some advantages/disadvantages:
+ Saves bandwidth as only the metadata is transferred
+ Less memory consumption for GCM as the client cache only stores the metadata
- Additional API server watch for resource --> additional watch cache for resource

Looking at this new reconciler, I'm not sure if the memory and bandwidth are really an issue since the amount of `gardener.cloud/role` secrets in the seed namespaces is rather small. Hence, I'd rather use an secret informer for `gardener.cloud/role` for ALL namespaces instead of haven one secret informer for the `garden` namespace and a metadata informer for other namespaces.
Any opinions?

Reply:
I reverted it back to use the secret informer and as discussed moved it to the GCM factory, so that the ref controller can also make use of the same informer (0124670).